foursight_core/checks/ecs_recovery_check.py

```diff
@@ -36,12 +39,19 @@ def rollback_application_version(connection, **kwargs):
         matching_cluster = matching_cluster[0]
         services = ecs_utils.list_ecs_services(cluster_name=matching_cluster)
         deploy_is_active = ecs_utils.service_has_active_deployment(cluster_name=matching_cluster, services=services)
-        # check previous run
-        if deploy_is_active:
+        # check previous run, checking that ES is reachable
+        if deploy_is_active and (connection.test_es_connection()):
             last_result = check.get_primary_result()
             prior_result = check.get_closest_result(diff_mins=30)
-            # 2 prior warns indicate deployment failure
-            if last_result.status == 'WARN' and prior_result.status == 'WARN':
+            # set to warn if either of previous results don't exist
+            if not last_result or not prior_result:
+                check.status = 'WARN'
+                check.summary = f'Detected that deployment is still running - after 3 consecutive checks, image will' \
+                                f' be rolled back on cluster {matching_cluster}'
```

Comment:
A few cosmetic mods here. If there is a sentence break, I prefer to break the line there. I don't really like \ to mark a continuation line so changed to parens. I also didn't get the hyphen in the middle of the sentence.
Untested:
```suggestion
                check.summary = (f'Detected that deployment is still running after 3 consecutive checks.'
                                 f' Image will be rolled back on cluster {matching_cluster}.')
```